Source/Core/AudioCommon/SurroundDecoder.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <limits>
+
+#include "AudioCommon/SurroundDecoder.h"
+#include "FreeSurround/FreeSurroundDecoder.h"
+
+namespace AudioCommon
+{
+SurroundDecoder::SurroundDecoder(uint32_t sample_rate, uint32_t frame_block_size)
+{
+  m_sample_rate = sample_rate;
+  m_frame_block_size = frame_block_size;
+  m_fsdecoder = std::make_shared<DPL2FSDecoder>();
+  m_fsdecoder->Init(cs_5point1, m_frame_block_size, m_sample_rate);
+}
+
+void SurroundDecoder::Clear()
+{
+  m_fsdecoder->flush();
+  m_decoded_fifo.clear();
+}
+
+// Currently only 6 channels are supported.
+uint32_t SurroundDecoder::QueryFramesNeededForSurroundOutput(uint32_t output_frames)
+{
+  if (m_decoded_fifo.size() < output_frames * SURROUND_CHANNELS)
+  {
+    // Output stereo frames needed to have at least the desired number of surround frames
+    uint32_t frames_needed =
+        output_frames - static_cast<uint32_t>(m_decoded_fifo.size()) / SURROUND_CHANNELS;
+    return frames_needed + m_frame_block_size - frames_needed % m_frame_block_size;
+  }
+  else
```

Comment:
`else` can be omitted, since the `if` contains a return.

i.e.

```cpp
if (...)
{
  return ...;
}
else
{
  return ...;
}
```

can be

```cpp
if (...)
{
  return ...;
}

return ...;
```